packages/algoliasearch-helper/src/algoliasearch.helper.js

```diff
@@ -1578,6 +1592,78 @@ AlgoliaSearchHelper.prototype._search = function (options) {
   return undefined;
 };
 
+/**
+ * Perform the underlying queries
+ * @private
+ * @param {boolean} [options.onlyWithDerivedHelpers=false] if true, only the derived helpers will be queried
+ * @return {undefined} does not return anything
+ * @fires search
+ * @fires result
+ * @fires error
+ */
+AlgoliaSearchHelper.prototype._runComposition = function () {
+  var state = this.state;
+  var states = [];
+  var mainQueries = [];
+  const thisCompositionID = this.compositionID;
+
+  var derivedQueries = this.derivedHelpers.map(function (derivedHelper) {
+    var derivedState = derivedHelper.getModifiedState(state);
+    var derivedStateQueries = thisCompositionID
+      ? requestBuilder._getCompositionQueries(thisCompositionID, derivedState)
+      : [];
+
+    states.push({
+      state: derivedState,
+      queriesCount: derivedStateQueries.length,
+      helper: derivedHelper,
+    });
+
+    derivedHelper.emit('search', {
+      state: derivedState,
+      results: derivedHelper.lastResults,
+    });
+
+    return derivedStateQueries;
+  });
+
+  var queries = Array.prototype.concat.apply(mainQueries, derivedQueries);
+
+  var queryId = this._queryId++;
+  this._currentNbQueries++;
+
+  if (!queries.length) {
+    return Promise.resolve({ results: [] }).then(
+      this._dispatchAlgoliaResponse.bind(this, states, queryId)
+    );
+  }
+
+  if (queries.length > 1) {
+    throw new Error('Only one query is allowed when using a composition.');
+  }
+
+  var query = queries[0];
+
+  // TODO: delete once allowed in the Composition API
+  delete query.params.maxValuesPerFacet;
+  delete query.params.facets;
+
+  const { compositionID, params } = query;
+  try {
+    this.client
+      .search({ compositionID, requestBody: { params } })
```

Comment:
this is temporary I presume?

Reply:
no it isn't.
I named the method on the composition API client `search`

Reply:
I pushed an update so that the query object is correctly build from the start (uglify was complaining)